custom_components/car_wash/binary_sensor.py

```diff
@@ -157,7 +160,23 @@ async def async_update(self):
         tmpu = self.hass.config.units.temperature_unit
         temp = wdata.attributes.get(ATTR_WEATHER_TEMPERATURE)
         cond = wdata.state
-        forecast = wdata.attributes.get(ATTR_FORECAST)
+
+        daily_response = await self.hass.services.async_call(
+            WEATHER_DOMAIN,
+            SERVICE_GET_FORECASTS,
+            {"entity_id": self._weather_entity, "type": "daily"},
+            blocking=True,
+            return_response=True,
+        )
+        if not (isinstance(daily_response, dict)):
+            self._attr_is_on = None
+            raise HomeAssistantError("daily_response is empty.")
```

Comment:
This error message is not clear to the user if he sees it in the log.  The user does not know what daily_response is.

Reply:
fixed it with new commit, by just using the original message
"Can't get forecast data! Are you sure it's the weather provider?"

by setting `forecast = None`

Reply:
The option is suitable, but it will be better if the error message is more unique, so that it is also more clear to developers about where the failure actually occurred.